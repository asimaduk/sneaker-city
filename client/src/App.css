.Product-Div {
  width: 100%;
  height: 100%;
  position: relative;
  color: white;
  display: block;
}

.Product-Image {
  width: 100%;
  min-height: 150px;
  background-color: #cccccc;
}

.Grid {
  padding-right: 8px;
  padding-left: 8px;
}

.Product-Details {
  height: 22%;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
}

.Product-Price {
  position: absolute;
  top: 0;
  left: 0;
  height: 50px;
  width: 100px;
  background-color: black;
  display: flex;
  align-items: center;
  justify-content: center;
  border-bottom-right-radius: 5px;
}

.Product-Details-Price, .Product-Details-ReleaseDate {
  color: black;
  text-align: center;
  text-transform: uppercase;
  font-size: 15px;
  font-weight: 500;
}

.Product-Details-Sizes {
  padding-left: 16px;
  padding-right: 30px;
  margin-top: 30px;
  margin-bottom: 0;
  font-weight: 500
}

.Product-Wrapper {
  padding-bottom: 20px;
  padding-top: 20px;
  text-align: center;
}

.Product-Name {
  text-align: center;
  color: #6d6d6d;
  text-transform: uppercase;
  font-size: 14px;
  line-height: 1.714285714285714;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-weight: 400;
  letter-spacing: .5px;
  margin: 0;
  padding: 0;
  border: 0;
}

.Product-Desc {
  color: black;
  text-align: center;
  text-transform: uppercase;
  font-size: 20px;
  line-height: 1.714285714285714;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  letter-spacing: .5px;
  display: block;
  margin: 0;
  padding: 0;
  border: 0;
}

.Product-ReleaseDate {
  color: #000000;
  padding-top: 3px;
  margin-block-start: 0;
  text-transform: uppercase;
  font-size: 14px;
  font-weight: 500;
  line-height: 1.5;
}

.Shopping-Cart-Icon {
  margin-right: 10px;
}

.Cart-Link {
  color: white;
  text-decoration: none
}

.Cart-Div {
  display: flex;
  align-items: center;
  justify-content: center;
}

.Cart-Item-Name {
  color: #6d6d6d;
  font-size: 16px;
  margin: 0
}

.Cart-Item-Desc {
  color: #000000;
  font-size: 18px;
  font-weight: 500;
  margin: 0
}

.Cart-Item-Size {
  font-size: 13px;
  color: grey;
  margin: 0
}