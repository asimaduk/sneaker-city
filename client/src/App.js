import React, { Component } from 'react';

import NavBar from './components/NavBar';
import ProductList from './components/ProductList';

import ProductDetails from './routes/ProductDetails';
import Cart from './routes/Cart';
import CheckOut from './routes/CheckOut';
import NotFound from './routes/NotFound';

import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';

class App extends Component {
    constructor(){
        super();
    
        this.state = {
            cartItems: []
        }

        //adding functions (cart) to the global object window, so they can be accessed from 
        // any component

        window.addToCart = (item) => {
            let temp = this.state.cartItems;
            temp.push(item);

            this.setState({cartItems: temp});
        }

        window.removeFromCart = (item) => {
            const temp = this.state.cartItems.filter(product => {
                if((product.id === item.id) && (product.size === item.size)){
                    return false;
                }
                
                return true;
            });
      
            this.setState({cartItems: temp});
        }

        window.getCart = () => {
            return this.state.cartItems;
        }
      
        window.emptyCart = () => {
            this.setState({cartItems: []})
        }
    }

    render(){
        const { cartItems } = this.state;
    
        return (
            <Router>
                <div>
                    <NavBar cartItemsCount={cartItems.length}/>
                    
                    <Switch>
                        <Route path="/" exact component={ProductList} />
                        <Route path="/products/:brand/:id" component={ProductDetails}/>
                        <Route path="/cart" component={Cart} />
                        <Route path="/checkout" component={CheckOut} />
                        <Route path="*" component={NotFound} />
                    </Switch>
                </div>
            </Router>
        )
    }
}

export default App;