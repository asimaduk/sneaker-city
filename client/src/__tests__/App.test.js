import React from 'react';
import ReactDOM from 'react-dom';
import { create } from 'react-test-renderer';

import App from '../App';

describe('App', () => {
    test('renders without crashing', () => {
        const component = create(<App />); 
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
})
