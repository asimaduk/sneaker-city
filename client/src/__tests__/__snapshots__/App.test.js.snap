// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Tests renders without crashing 1`] = `
<div>
  <div>
    <header
      className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <div
          className="MuiGrid-root MuiGrid-container"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11"
            style={
              Object {
                "paddingTop": "3px",
              }
            }
          >
            <h6
              className="MuiTypography-root MuiTypography-h6"
              style={
                Object {
                  "fontWeight": "500",
                }
              }
            >
              Sneaker City
            </h6>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
          >
            <a
              className="Cart-Link"
              href="/cart"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
               
              0
            </a>
          </div>
        </div>
      </div>
    </header>
  </div>
  <div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "height": 300,
          "justifyContent": "center",
        }
      }
    >
      <div
        className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
